packages/app-mobile/components/NotesBar.tsx

```diff
@@ -0,0 +1,282 @@
+import * as React from 'react';
+import { View, Text, TextInput, TouchableOpacity, FlatList } from 'react-native';
+import { State } from '@joplin/lib/reducer';
+import { themeStyle } from './global-style';
+const { connect } = require('react-redux');
+const Icon = require('react-native-vector-icons/Ionicons').default;
+const { _ } = require('@joplin/lib/locale');
+import { Style } from './global-style';
+import Note from '@joplin/lib/models/Note';
+import NotesBarListItem from './NotesBarListItem';
+import Folder from '@joplin/lib/models/Folder';
+import useStyles from './useStyles';
+
+interface Props {
+    themeId: number;
+	items: any[];
+	todoCheckbox_change: (checked: boolean)=> void;
+	selectedFolderId: string;
+	activeFolderId: string;
+	dispatch: any;
+	selectedNoteId: string;
+	toggleNotesBar: ()=> void;
+	settings: any;
+}
+
+function NotesBarComponent(props: Props) {
+
+	const [notes, setNotes] = React.useState<any[]>(props.items);
```

Comment:
Please don't do that, just use props.items (or props.notes) directly. Having to manually sync the state and props causes a lot of troubles long term. Think about your component - if you somehow needs a view of the notes, then use useMemo to derive from the props the data that you need.

Reply:
Noted, thank you.

Reply:
I'll still need to use state, so that I can update the notes, with the results of a search query. Using a regular value won't update the UI. I don't know how I can do this with useMemo.